include/rive/event_type.hpp

```diff
@@ -0,0 +1,12 @@
+#ifndef _RIVE_EVENT_TYPE_HPP_
+#define _RIVE_EVENT_TYPE_HPP_
+namespace rive {
+    enum class EventType : int {
```

Comment:
Let's add some comments here:

Things I'm wondering about...
- if the mouse just moves, which event(s) might fire?
- does up/down strictly just mean the button changed state (i.e. no movement)?
- are up/down always "nested" inside enter/exit?
- does updateHover mean positionChanged, or is it in any way related to the other 4?